test/core/handshake/test_handshake.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright 2025 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "test/core/handshake/test_handshake.h"
+
+#include <memory>
+
+#include "src/core/handshaker/handshaker.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/event_engine/channel_args_endpoint_config.h"
+#include "src/core/lib/event_engine/memory_allocator_factory.h"
+#include "src/core/lib/event_engine/tcp_socket_utils.h"
+#include "src/core/lib/iomgr/event_engine_shims/endpoint.h"
+#include "src/core/lib/resource_quota/resource_quota.h"
+#include "test/core/event_engine/fuzzing_event_engine/fuzzing_event_engine.h"
+
+using ::grpc_event_engine::experimental::ChannelArgsEndpointConfig;
+using ::grpc_event_engine::experimental::EventEngine;
+using ::grpc_event_engine::experimental::FuzzingEventEngine;
+using ::grpc_event_engine::experimental::grpc_event_engine_endpoint_create;
+using ::grpc_event_engine::experimental::MemoryQuotaBasedMemoryAllocatorFactory;
+using ::grpc_event_engine::experimental::ResolvedAddressMakeWild4;
+
+namespace grpc_core {
+
+namespace {
+void Handshake(std::unique_ptr<EventEngine::Endpoint> endpoint,
+               const ChannelArgs& channel_args,
+               absl::optional<absl::StatusOr<ChannelArgs>>* output) {
+  auto handshake_mgr = MakeRefCounted<HandshakeManager>();
```

Comment:
This is creating a handshake manager but not actually adding any handshakers to it, so I think the handshake will always be a no-op.

I think we need to do something like what we do on client and server side to add handshakers:

https://github.com/grpc/grpc/blob/7bf13efd18620b3013ff8f67ba4ebeeb04e1c917/src/core/ext/transport/chttp2/client/chttp2_connector.cc#L116

https://github.com/grpc/grpc/blob/7bf13efd18620b3013ff8f67ba4ebeeb04e1c917/src/core/ext/transport/chttp2/server/chttp2_server.cc#L1018

Reply:
Yeah you're right... I thought that was too easy
Fixed.